Packs/QuestKace/Integrations/QuestKace/QuestKace.py

```diff
@@ -0,0 +1,549 @@
+
+
+def test_module(client: Client, args=None) -> Tuple[str, dict, dict]:
+    response = client.machines_list_request()
+    list_machines_res = response.get('Machines')
+    if list_machines_res:
+        return 'ok', {}, {}
+    else:
+        return 'Test failed', {}, {}
+
+
+def get_machines_list_command(client, args) -> Tuple[str, dict, dict]:
+    response = client.machines_list_request()
+    raw_response = response.get('Machines')
+    context = parse_response(raw_response)
+    human_readable_markdown = tableToMarkdown('Quest Kace Machines', context, removeNull=True)
+    context = {
+        'QuestKace.Machines(val.ID === obj.ID)': context
+    }
+    return human_readable_markdown, context, raw_response
+
+
+def get_assets_list_command(client, args) -> Tuple[str, dict, dict]:
+    response = client.assets_list_request()
+    raw_response = response.get('Assets')
+    context = parse_response(raw_response)
+    human_readable_markdown = tableToMarkdown('Quest Kace Assets', context, removeNull=True)
+    context = {
+        'QuestKace.Assets(val.ID === obj.ID)': context
+    }
+    return human_readable_markdown, context, raw_response
+
+
+def get_queues_list_command(client, args) -> Tuple[str, dict, dict]:
+    response = client.queues_list_request()
+    raw_response = response.get('Queues')
+    context = parse_response(raw_response)
+    human_readable_markdown = tableToMarkdown('Quest Kace Queues', context, removeNull=True)
+    context = {
+        'QuestKace.Queues(val.ID === obj.ID)': context
+    }
+    return human_readable_markdown, context, raw_response
+
+
+def get_queues_fields_list_command(client, args) -> Tuple[str, dict, dict]:
+    queue_number = args.get('queue_number')
+    response = client.queues_list_fields_request(queue_number)
+    raw_response = response.get('Fields')
+    context = parse_response(raw_response)
+    human_readable_markdown = tableToMarkdown(f'Quest Kace Queue {queue_number} Fields', context, removeNull=True)
+    context = {
+        'QuestKace.Queues.Fields(val.JsonKey === obj.JsonKey)': context
+    }
+    return human_readable_markdown, context, raw_response
+
+
+def get_tickets_list_command(client, args) -> Tuple[str, dict, dict]:
+    shaping_fields = args.get("custom_fields")
+    response = client.tickets_list_request(shaping_fields)
+    raw_response = response.get('Tickets')
+    context = parse_response(raw_response)
+    human_readable_markdown = tableToMarkdown(f'Quest Kace Tickets', context, removeNull=True)
+    context = {
+        'QuestKace.Tickets(val.ID === obj.ID)': context
+    }
+    return human_readable_markdown, context, raw_response
+
+
+def create_ticket_command(client, args) -> Tuple[str, dict, dict]:
+    hd_queue_id = args.get('queue_id')
+    custom_fields = args.get('custom_fields')
+    if not hd_queue_id and not custom_fields:
+        raise DemistoException("queue id is a mandatory value, please add it to your request")
```

Comment:
No , the value hd_queue_id is mandatory in every create request.
If it is a custom request = custom_fields exist- then the user will need to add it in the custom_field.
Otherwise , it is mandatory for building the request by the integration.